cmd/internal/cli/actions_linux.go

```diff
@@ -124,20 +75,39 @@ func execStarter(cobraCmd *cobra.Command, image string, args []string, name stri
 
 	syscall.Umask(0022)
 
-	starter := buildcfg.LIBEXECDIR + "/singularity/bin/starter-suid"
-
 	engineConfig := singularityConfig.NewConfig()
-
-	configurationFile := buildcfg.SYSCONFDIR + "/singularity/singularity.conf"
-	if err := config.Parser(configurationFile, engineConfig.File); err != nil {
-		sylog.Fatalf("Unable to parse singularity.conf file: %s", err)
+	engineConfig.File, err = getFileConfig()
+	if err != nil {
+		sylog.Fatalf("%s", err)
 	}
 
 	ociConfig := &oci.Config{}
 	generator := generate.Generator{Config: &ociConfig.Spec}
 
 	engineConfig.OciConfig = ociConfig
 
+	allowedSuid := engineConfig.File.AllowSetuid
```

Comment:
`allowSuid` instead of `allowedSuid` (just like the original field)

```suggestion
	allowSuid := engineConfig.File.AllowSetuid
```